sublime_linter.py

```diff
@@ -332,22 +316,25 @@ def join_msgs(cls, line_dict, we_count, show_count=False):
                 <div>{messages}</div>
             '''
 
-        template = "{linter}: {code} - {escaped_msg}"
-        template_no_code = "{linter}: {escaped_msg}"
+        tmpl_with_code = "{linter}: {code} - {escaped_msg}"
+        tmpl_sans_code = "{linter}: {escaped_msg}"
 
         all_msgs = ""
         for error_type in WARN_ERR:
-            count = we_count[error_type]
             heading = error_type
-            error_type_msgs = []
-            msg_list = line_dict.get(error_type)
+            filled_templates = []
+            msg_list = [e for e in errors if e["error_type"] == error_type]
 
             if not msg_list:
                 continue
+
+            msg_list = sorted(msg_list, key=lambda x: (x["start"], x["end"]))
+            count = len(msg_list)
+
             for item in msg_list:
-                item["escaped_msg"] = html.escape(item["msg"], quote=False)
-                template_to_fill = template if item.get('code') else template_no_code
-                error_type_msgs.append(template_to_fill.format(**item))
+                msg = html.escape(item["msg"], quote=False)
```

Comment:
Message sanitation here again. Maybe we should put that near the `rstrip` we moved previously.
(In another PR.)

Reply:
I think the html.escape should be done just-in-time. It’s not just sanitation of the data, it’s a transformation of the string specifically for this situation. It’s 👍🏻 as is IMO.